### PR TITLE
processChanges use new API that spans all revisions when extracting changes

### PR DESCRIPTION
**Pull request by MichaelBelousov**
- annotate large branch synchronization test
- use new multi-changeset change extraction instead of doing it on each changeset like before which is actually not technically supported
- update tests to extract changes when it's valid, before making further changes to the database
- there is an accompanying native PR 241036

### CODE REVIEW

**Review by ColinKerr — APPROVED** [review]

**Comment by MichaelBelousov** [comment]
waiting for approval of the native addon changes

**Comment by MichaelBelousov** [comment]
abandoning in favor of a deeper refactor

**Pull request closed without merging**
